src/connector/src/parser/csv_parser.rs

```diff
@@ -76,7 +95,7 @@ impl CsvParser {
     fn parse_string(dtype: &DataType, v: String) -> AccessResult {
         let v = match dtype {
             // mysql use tinyint to represent boolean
-            DataType::Boolean => (parse!(v, i16)? != 0).into(),
+            DataType::Boolean => parse!(v, BooleanParser)?.into(),
```

Comment:
The `expected` field in `TypeError` will be `"BooleanParser"` then 😕, which does not look good. What about simply expanding it?

Reply:
fixed